charts/opentelemetry-operator/values.yaml

```diff
@@ -339,7 +339,8 @@ clusterRole:
 
 affinity: {}
 tolerations: []
-nodeSelector: {}
+nodeSelector:
+  kubernetes.io/os: linux
```

Comment:
this feels like maybe too big an assumption IMO. although it should always be the case that a node has this, i'm not sure it's a full requirement that it does. From the docs it's stated to be a [well known label](https://kubernetes.io/docs/reference/labels-annotations-taints/#kubernetes-io-os), but i'm not sure that means it's always present. Is there something more concrete that ensures we can rely on this always?

Reply:
definitely possible i'm incorrect here, and happy to be proven wrong.

Reply:
The os label is added by the kubelet to the node object when it is being readied for registration with the API - [ref](https://github.com/kubernetes/kubernetes/blob/v1.32.0/pkg/kubelet/kubelet_node_status.go#L309). Imo, it should always exists unless using some non-std kubelet.

If we are still hesitant about the nodeselector, another option would be to set a best effort affinity rule like following

```
spec:
  affinity:
    nodeAffinity:
      preferredDuringSchedulingIgnoredDuringExecution:
      - weight: 1
        preference:
          matchExpressions:
          - key: kubernetes.io/os
            operator: In
            values:
            - linux
```

Reply:
ah thats the exact ref i need to make me a bit more confident here, thanks for pointing me there!